src/pages/login/login.js

```diff
@@ -0,0 +1,163 @@
+import React, { useEffect, useState } from 'react';
+import { Button, TextField } from '@material-ui/core';
+import { Link } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { useStyles } from './login.styles';
+import {
+  placeholders,
+  OR_TEXT,
+  LOGIN_FORM_LABEL,
+  LOGIN_USER_DATA,
+  formRegExp,
+  LOGIN_USER_ERROR,
+  FORGOT_PASSWORD,
+  REGISTER_PROPOSAL
+} from '../../configs';
+import { loginUser } from '../../redux/user/user.actions';
+import { endAdornment } from '../../utils/eyeToggle';
+import { Loader } from '../../components/Loader/loader';
+
+const Login = () => {
+  // VALUES
+  const [user, setUser] = useState(LOGIN_USER_DATA);
+
+  // VALIDATE
+  const [shouldValidate, setShouldValidate] = useState(false);
+  const [emailValidated, setEmailValidated] = useState(false);
+  const [passwordValidated, setPasswordValidated] = useState(false);
+  const [allFieldsValidated, setAllFieldsValidated] = useState(false);
+
+  // SHOW PASSWORDS
+  const [showPassword, setShowPassword] = useState(true);
+
+  // HANDLERS
+  const handleChange = (event, setValid, regExp) => {
+    const input = event.target.value;
+    const inputName = event.target.name;
+    setUser({ ...user, [inputName]: input });
+    if (input.match(regExp)) {
+      setValid(true);
+    } else {
+      setValid(false);
+    }
+  };
+
+  const handleLogin = async () => {
+    setShouldValidate(true);
+    if (allFieldsValidated) {
+      try {
```

Comment:
Why do you wrap call of the `dispatch` it in `try {...} catch` block

Reply:
removed